File: 3-CODING-BLOCK/1-HASHING/4-LONGES-SUBSTRING.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        mp={}
        ans=0
        j=0
        for i in range(len(s)):
            if(s[i] in mp):
                j=max(j,mp[s[i]]+1)
            mp[s[i]]=i
            ans=max(ans,i-j+1)
        return ans
        
       

File: 3-CODING-BLOCK/1-HASHING/4-LONGES-SUBSTRING.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function (s) {
  let ans = 0;
  let mp = {};
  let j = 0;
  for (let i = 0; i < s.length; i++) {
    if (s[i] in mp) {
      j = Math.max(j, mp[s[i]] + 1);
    }
    mp[s[i]] = i;
    ans = Math.max(ans, i - j + 1);
  }
  return ans;
};
